Objects/bytesobject.c

```diff
@@ -2593,8 +2597,11 @@ bytes_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     if (PyIndex_Check(x)) {
         size = PyNumber_AsSsize_t(x, PyExc_OverflowError);
         if (size == -1 && PyErr_Occurred()) {
-            if (PyErr_ExceptionMatches(PyExc_OverflowError))
+            if (PyErr_ExceptionMatches(PyExc_OverflowError)) {
+                PyErr_SetString(PyExc_OverflowError,
```

Comment:
should I open a sub-issue of bpo-29833 for this and for `bytearray_init()` (in Objects/bytearrayobject.c)?